static/app/components/button.tsx

```diff
@@ -219,8 +219,27 @@ function BaseButton({
   const accessibleLabel =
     ariaLabel ?? (typeof children === 'string' ? children : undefined);
 
-  const useButtonTracking = HookStore.get('react-hook:use-button-tracking')[0];
-  const buttonTracking = useButtonTracking?.({
+  const useButtonTrackingLogger = () => {
+    const hasAnalyticsDebug = window.localStorage?.getItem('DEBUG_ANALYTICS') === '1';
+    if (!analyticsEventName || !hasAnalyticsDebug) {
```

Comment:
why do we need the check for `analyticsEventName` here? is it just so we track this if they are sending to amplitude, and not just reload?

Reply:
It's because `useButtonTracking` is called regardless of wether the fields are set or not. So a lot of empty log messages can appear. 

I'll double-check though, and probably make a change so that we're: a) checking both the fields (either name or key can be set)
b) make it clearer why we are checking, with a variable name for example

Reply:
depends on how we want to use this debugging, if we check that either of the fields are set, then we end up in a scenario where if you don't want to change the default name, you won't see anything in the logs (since you don't need to pass either the key or name in). if we want to restrict it to you need to set a name/key then it becomes less spammy but could filter out some of the default tracks which maybe you want to see

Reply:
so without the guard we'll get loads of empty `buttonAnalyticsEvent`'s in the console:

![SCR-20230926-npxk](https://github.com/getsentry/sentry/assets/187460/40e2239d-f61c-41d3-a40c-91ec38b5073b)

Reply:
In our getsentry hook we do check for eventName before logging: https://github.com/getsentry/getsentry/blob/404227b821072d3fd40bfda4953140cad207e86e/static/getsentry/gsApp/utils/rawTrackAnalyticsEvent.tsx#L216-L233

and there's another handler and system for the eventKeys which i didn't fully track down, but i assume it's also got an if-statement as well.

I updated the code here to check for either.

Reply:
> and there's another handler and system for the eventKeys which i didn't fully track down

so the one i didnt' track down, it does log using a generated name.

Updated the code to log whenever any of the three `analyticsEventName || analyticsEventKey || analyticsParams` things are passed, because you're sending custom stuff, so we should let you see it!